src/ui/AutoCompleteSuggest.ts

```diff
@@ -23,12 +23,20 @@ import { AppHelper } from "../app-helper";
 
 function suggestWords(words: Word[], query: string, max: number): Word[] {
   return Array.from(words)
-    .filter((x) => x.value !== query)
```

Comment:
I remove this to L27, because I think it will cost more time when words list in very long, in my case, 2w words 😂

Reply:
As you said, it looks more faster, thanks 👍 

![image](https://user-images.githubusercontent.com/9500018/143583764-583efcd3-f456-48fe-8ef1-e47ad1abc41b.png)

Reply:
A similar improvement can be done at https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/blob/main/src/ui/AutoCompleteSuggest.ts#L41.

This will iterate the words again.
A better ways is just use a list, and iterate the word list one time, put valid candidate to the list.

Reply:
> This will iterate the words again.

What do you mean? 🤔

Reply:
> 

After you merge the code, the code line number is not right now.
It's now https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/blob/main/src/ui/AutoCompleteSuggest.ts#L49.

we may remove this line for performance, because it will iterate the long list again.

```javascript
.filter((x) => x.value !== undefined)
```

Reply:
Thank you for your reply :)  I understood.

![image](https://user-images.githubusercontent.com/9500018/143670991-a911ff75-e301-4e12-a596-8c79ccba3b5d.png)

On the other hand, It might be slow in the 10000case instead of 10000000case.

![image](https://user-images.githubusercontent.com/9500018/143671067-952ab4eb-985d-498f-8373-7fd90fcdf83e.png)

So I prior to readability for now.